### PR TITLE
Prepare documentation using site in GitHub Pages

### PR DESCRIPTION
**Pull request by mroderick**
This is a PR for **master**, it implements a new documentation site using GitHub Pages (well Jekyll underneath).

This is the squashed version of my work-in-progress branch. I can't imagine anyone being interested in seeing all the detours I took to get here... if I am wrong, please speak up and I'll post a link to it.

I (now) think we can do everything that we need to with GitHub Pages, so there's really no reason to complicate things by having yet another system to manage.

---

There's still a lot of work left to do on the new documentation website, however I think we should merge this branch, as it will:

* allow me to finish configuring GitHub Pages, so we can see that the site actually works there
* allow others to contribute to finishing the documentation site

If this PR is merged, I will update https://github.com/sinonjs/sinon/issues/991 to track the remaining work. Specifically, I will (as a minimum) add these tasks.

- [x] Script for copying `docs/release-source/` into `docs/_releases/` as a new release, with the new release's id.
- [x] Script for setting `current_release` and/or `next_release` values in `docs/_config.yml` whenever a new release is created
- [x] Update Changelog.txt and AUTHORS whenever a new release is cut
- [x] Create a Jekyll collection for how to articles
- [ ] Create a Jekyll collection for packaged versions for download (http://sinonjs.org/download/)
- [ ] Script for copying the packaged release from `pkg/` to packaged versions, when a new release is cut
- [ ] Script for copying `Changelog.txt` into the documentation site whenever a new release is cut
- [ ] Link to the Changelog
- [ ] Add prominent links to source and issues on GitHub
- [ ] Go through the documentation of `v1.17.6` and remove any changes that have been made for `v2.0.0`. Perhaps we can re-import the current documentation from `sinonjs/sinon-docs`

I will try to convince the creator of the new favicon to create a better design and some illustrations for the site.

---

#### To verify

1. Check out this branch
1. `cd docs`
1. `gem install bundler`
1. `bundle install`
1. `bundle exec jekyll serve`
1. Navigate to http://localhost:4000/
1. Click around, and see if you can find any errors

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
 
-You'll need a working installation of [git-extras](https://github.com/tj/git-extras) for this.
+You will need a working installation of [git-extras](https://github.com/tj/git-extras) for this.
 
-## Update Changelog.txt
+The release process is mostly automated, here is a brief overview of the steps
 
-Compile interesting highlights from [`git changelog`](https://github.com/tj/git-extras/blob/master/Commands.md#git-changelog) into Changelog.md
+1. `npm version [keyword]`
+    - Updates `Changlelog.txt` - you will need to edit this
```
Comment (fatso83) [inline RELEASE.md]: Typo

File: docs/CONTIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Contribruting to documentation
```
Comment (fatso83) [inline docs/CONTIBUTING.md]: typo. also the name of the file is missing an r.

**Comment by mantoni** [comment]
Just played around with the website a little. Looks really good. I like it 👍 

The only thing that I found are some funny quotes in APIs. I guess that is macOS being clever about quotations. Especially the `matchers` page has some and in the `spy` and `stub` pages some double quotes have a leading backslash.

Solid work Morgan. Very simple to get started. Thanks!

File: docs/release-source/release/spies.md [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+
+Resets the state of a spy.
+
+
+#### `spy.restore()`
+
+Replaces the spy with the original method. Only available if the spy replaced an existing method.
+
+
+#### `spy.printf(format string\", [arg1, arg2, ...])`
```
Comment (mantoni) [inline docs/release-source/release/spies.md]: I guess it should be `spy.printf("format string", [arg1, arg2, ...])`

File: RELEASE.md (hunk 1 above)
Comment (mroderick) [inline RELEASE.md]: I think that was fixed in a later commit, I can't find it in the source

Comment (mroderick) [inline RELEASE.md]: Bah, looking in the wrong project ..

File: docs/CONTIBUTING.md (hunk 2 above)
Comment (mroderick) [inline docs/CONTIBUTING.md]: Fixed

File: docs/_releases/v1.17.6/matchers.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+Requires the value to be a `Number`.
+
+
+#### `sinon.match.string`
+
+Requires the value to be a `String`.
+
+
+#### `sinon.match.object"
```
Comment (mroderick) [inline docs/_releases/v1.17.6/matchers.md]: Fixed

File: docs/release-source/release/spies.md (hunk 3 above)
Comment (mroderick) [inline docs/release-source/release/spies.md]: Fixed

**Comment by mroderick** [comment]
@mantoni @fatso83 I've addressed all comments so far. I would prefer to merge all of this as one commit, so we don't end up with all the `Review:` commits in history

**Comment by fatso83** [comment]
Crap. Sorry that I held this up. I had no idea I was the one stalling it. And now it seems there have arisen some conflicts in the docs ...

**Comment by mroderick** [comment]
No worries, I wasn't sure myself. I'll rebate it tomorrow

**Comment by mroderick** [comment]
Rebase even. Damn autocorrect

**Comment by mroderick** [comment]
Rebased. I think this is good to go now

**Comment by mroderick** [comment]
Thanks!

**Pull request merged**